packages/react-admin/src/mui/field/ReferenceField.spec.js

```diff
@@ -54,14 +55,65 @@ describe('<ReferenceField />', () => {
         const linkElement = wrapper.find('WithStyles(Link)');
         assert.equal(linkElement.prop('to'), '/bar/123');
     });
+    it('should render a link to the Edit page of the related record when the resource contains slashes', () => {
+        const wrapper = shallow(
+            <ReferenceField
+                record={{ fooId: 123 }}
+                source="fooId"
+                referenceRecord={{ id: 123, title: 'foo' }}
+                reference="prefix/bar"
+                resource="prefix/foo"
+                basePath="/prefix/foo"
+                crudGetManyAccumulate={() => {}}
+            >
+                <TextField source="title" />
+            </ReferenceField>
+        );
+        const linkElement = wrapper.find('WithStyles(Link)');
+        assert.equal(linkElement.prop('to'), '/prefix/bar/123');
+    });
+    it('should render a link to the Edit page of the related record when the resource is named edit or show', () => {
+        let wrapper = shallow(
+            <ReferenceField
+                record={{ fooId: 123 }}
+                source="fooId"
+                referenceRecord={{ id: 123, title: 'foo' }}
+                reference="edit"
+                resource="show"
+                basePath="/edit"
```

Comment:
So we're in a list of `show` resources but the basepath is `/edit`? I'm not sure that makes sense, it should be `/show`